spec/models/audio_spec.rb

```diff
@@ -1,27 +1,84 @@
 require 'rails_helper'
 
 RSpec.describe Audio, type: :model do
-  before(:each) do
-    @uploader = User.create(uid: '123', nickname: 'louise')
-  end
+  it { is_expected.to belong_to(:uploader).class_name('User') }
+  it { is_expected.to validate_presence_of(:title) }
+  it { is_expected.to validate_length_of(:title).is_at_least(4) }
+  it { is_expected.to validate_presence_of(:filename) }
+
+  it do
+    create(:audio)
 
-  it "is not valid without a title"  do
-    audio = Audio.new(title: nil)
-    expect(audio).to_not be_valid
+    is_expected.to validate_uniqueness_of(:filename)
   end
 
-  it "is not valid when title is too short" do
-    audio = Audio.new(title: 'boo')
-    expect(audio).to_not be_valid
+  describe '#update_metadata' do
```

Comment:
@nakedsushi I tried to reproduce the shrine behavior with the file here in the test environment.
But without luck, so I just stubed and mocked a bunch of stuff here.